src/server/index.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Server Helpers for MCP Apps.
+ *
+ * @module server-helpers
+ */
+
+import {
+  RESOURCE_URI_META_KEY as _RESOURCE_URI_META_KEY,
+  McpUiResourceMeta,
+  RESOURCE_MIME_TYPE,
+} from "../app.js";
+import type {
+  McpServer,
+  ResourceMetadata,
+  ToolCallback,
+  ReadResourceCallback,
+} from "@modelcontextprotocol/sdk/server/mcp.js";
+import type { ToolAnnotations } from "@modelcontextprotocol/sdk/types.js";
+import type { ZodRawShape } from "zod";
+
+// Re-export SDK types for convenience
+export type { ResourceMetadata, ToolCallback, ReadResourceCallback };
+
+// Re-export for convenience
+export const RESOURCE_URI_META_KEY = _RESOURCE_URI_META_KEY;
+export { RESOURCE_MIME_TYPE };
+
+/**
+ * Tool configuration (same as McpServer.registerTool).
+ */
+export interface ToolConfig {
+  title?: string;
+  description?: string;
+  inputSchema?: ZodRawShape;
+  annotations?: ToolAnnotations;
+  _meta?: Record<string, unknown>;
+}
+
+/**
+ * MCP App Tool configuration for `registerAppTool`.
+ */
+export interface McpUiAppToolConfig extends ToolConfig {
+  _meta: {
+    // Soon: `ui: McpUiToolMeta;` (https://github.com/modelcontextprotocol/ext-apps/pull/131)
+
+    /**
+     * URI of the UI resource to display for this tool.
+     * This is converted to `_meta["ui/resourceUri"]`.
+     *
+     * @example "ui://weather/widget.html"
+     */
+    [RESOURCE_URI_META_KEY]?: string;
+  };
```

Comment:
I think this needs `[key: string]: unknown;`; otherwise, it will prevent adding other `_meta` values.

```suggestion
  _meta: {
    // Soon: `ui: McpUiToolMeta;` (https://github.com/modelcontextprotocol/ext-apps/pull/131)

    /**
     * URI of the UI resource to display for this tool.
     * This is converted to `_meta["ui/resourceUri"]`.
     *
     * @example "ui://weather/widget.html"
     */
    [RESOURCE_URI_META_KEY]?: string;
    [key: string]: unknown;
  };
```